pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -352,7 +353,7 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 	logger.Debug("Retrieving data to create intermediate secret.")
 	retriever := NewRetriever(r.dynClient, plan, sbr.Spec.EnvVarPrefix)
-	retrievedData, err := retriever.Retrieve()
+	retrievedData, retrievedCache, err := retriever.Retrieve()
```

Comment:
In the current design of the operator, `Retriever` should only fetch data specified by the `Plan` (from `Planner`), therefore, `Retreive` should only return one single set of data, not two. The idea is the later components won't have to figure out sources of data anymore, and act in a single view of the binding context data.